hook-emulator/c_wrappers/main.c

```diff
@@ -62,486 +29,114 @@ etxn_inf = {0, 0};
+    offset += 1;
 
-    // If result code is 0 return EMISSION_FAILURE.
-    if (ret < 0 || (data_len - 1) != TRANSACTION_HASH_LEN)
+    txn->memos.list = (struct Memo *)malloc(sizeof(struct Memo) * txn->memos.memo_count);
+    uint8_t *data_ptr = buffer + offset;
+    for (size_t i = 0; i < txn->memos.memo_count; i++)
     {
-        write_ptr = 0;
-        return EMISSION_FAILURE;
-    }
-
-    // Populate the received transaction hash to the write pointer.
-    memcpy(write_ptr, res, TRANSACTION_HASH_LEN);
-    return 0;
-}
-
-int64_t state_set(uint32_t read_ptr, uint32_t read_len, uint32_t kread_ptr, uint32_t kread_len)
-{
-    // Acording to the xrpl hook specs,
-    // If kread_len is greater than max key len or read len is greater than max data len, return TOO_BIG.
-    // If kread_len is 0, return TOO_SMALL.
-    if (kread_len > STATE_KEY_LEN || read_len > MAX_STATE_VAL_LEN)
-        return TOO_BIG;
-    else if (kread_len == 0)
-        return TOO_SMALL;
-
-    // Send the state set request.
-    const int len = 1 + STATE_KEY_LEN + read_len;
-    uint8_t buf[len];
-    // Populate the type header.
-    buf[0] = STATE_SET;
-    // Populate the state data.
-    memcpy(&buf[1], kread_ptr, kread_len);
-    // If kread_len is less than STATE_KEY_LEN, populate o's to the rest.
-    if (kread_len < STATE_KEY_LEN)
-        memset(&buf[1 + kread_len], 0, STATE_KEY_LEN - kread_len);
-    // Populate the data to rest.
-    memcpy(&buf[1 + STATE_KEY_LEN], read_ptr, read_len);
-    write_stdout(buf, len);
-
-    // Read the response from STDIN.
-    int8_t ret;
-    const int data_len = read_stdin(&ret, 1);
-
-    // Return the error code according to the return code.
-    if (ret == RET_OVERFLOW)
-        return TOO_BIG;
-    else if (ret == RET_UNDERFLOW)
-        return TOO_SMALL;
-    else if (ret < -1)
-        return -1;
-
-    return 0;
-}
-
-int64_t state(uint32_t write_ptr, uint32_t write_len, uint32_t kread_ptr, uint32_t kread_len)
-{
-    // Acording to the xrpl hook specs,
-    // If kread_len is greater than max key len, return TOO_BIG.
-    if (kread_len > STATE_KEY_LEN)
-        return TOO_BIG;
-
-    // Send the state set request.
-    const int len = 1 + kread_len;
-    uint8_t buf[len];
-    // Populate the type header.
-    buf[0] = STATE_GET;
-    // Populate the state key.
-    memcpy(&buf[1], kread_ptr, kread_len);
-    write_stdout(buf, len);
-
-    // Read the response from STDIN.
-    uint8_t data_buf[MAX_STATE_VAL_LEN];
-    const int data_len = read_stdin(data_buf, sizeof(data_buf));
-    const int ret = (int8_t)*data_buf;
-    const uint8_t *res = &data_buf[1];
-
-    if (ret < 0)
-    {
-        write_ptr = 0;
-        // Return the error code according to the return code.
-        if (ret == RES_NOT_FOUND)
-            return DOESNT_EXIST;
-        else if (ret == RET_OVERFLOW)
-            return TOO_BIG;
-        else if (ret == RET_UNDERFLOW)
-            return TOO_SMALL;
+        const uint8_t type_len = *data_ptr;
```

Comment:
Just a suggestion: what if we directly assign to `txn->memos.list[i].type_len` (Can do the same for format and data)

Reply:
Resolving as per discussion